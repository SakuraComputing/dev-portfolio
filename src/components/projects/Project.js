import React from 'react';
import ProjectItem from '../../components/projects/ProjectItem';

class Project extends React.Component {



    render() {

        const { projects, latestproject } = this.props;

        let projectItems;

        if(projects.length > 0) {
            projectItems = projects.map(project => (
                    <ProjectItem key={project.id} project={project}/>
                )
            );
        }

        return (
            <section className="latest section">
                <div className="section-inner">
                    <h2 className="heading">Latest Projects</h2>
                    <div className="content">

                        <div className="item featured text-center">
                            <h3 className="title"><a
                                href={latestproject.exampleURL}
                                target="_blank">{latestproject.title}</a></h3>
                            <p className="summary">{latestproject.synopsis}</p>
                            <div className="featured-image">
                                <a href={latestproject.exampleURL}
                                   target="_blank">
                                    <img className="img-responsive project-image"
                                         src={latestproject.imagePath} alt="project name"/>
                                </a>
                                <div className="ribbon">
                                    <div className="text">New</div>
                                </div>
                            </div>

                            <div className="desc text-left">
                                <p>{latestproject.description1}</p>
                                <p>{latestproject.description2}</p>
                            </div>
                            <a className="btn btn-cta-secondary"
                               href={latestproject.exampleURL}
                               target="_blank"><i className="fa fa-thumbs-o-up"></i> Demo</a>
                        </div>
                        <hr className="divider"/>

                        {projectItems}
                    </div>

                </div>
            </section>

        )
    }
}

export default Project;
