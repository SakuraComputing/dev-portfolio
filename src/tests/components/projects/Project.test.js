import React from 'react';
import { shallow } from 'enzyme';
import Project from '../../../components/projects/Project';

describe('Latest Project Section', () => {
    it('should render correctly', () => {
        const projects = jest.fn();
        const wrapper = shallow(<Project projects={projects}/>);
        expect(wrapper).toMatchSnapshot();
    });
});