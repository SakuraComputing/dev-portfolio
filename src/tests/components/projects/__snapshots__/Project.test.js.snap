// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Latest Project Section should render correctly 1`] = `
<section
  className="latest section"
>
  <div
    className="section-inner"
  >
    <h2
      className="heading"
    >
      Latest Projects
    </h2>
    <div
      className="content"
    >
      <div
        className="item featured text-center"
      >
        <h3
          className="title"
        >
          <a
            href="http://themes.3rdwavemedia.com/website-templates/responsive-bootstrap-theme-for-products-velocity/"
            target="_blank"
          >
            KickStarter - Velocity
          </a>
        </h3>
        <p
          className="summary"
        >
          A responsive Bootstrap template designed for digital products
        </p>
        <div
          className="featured-image"
        >
          <a
            href="http://themes.3rdwavemedia.com/website-templates/responsive-bootstrap-theme-for-products-velocity/"
            target="_blank"
          >
            <img
              alt="project name"
              className="img-responsive project-image"
              src="assets/images/projects/project-featured.png"
            />
          </a>
          <div
            className="ribbon"
          >
            <div
              className="text"
            >
              New
            </div>
          </div>
        </div>
        <div
          className="desc text-left"
        >
          <p>
            You can promote your main project here. Suspendisse in tellus dolor. Vivamus a tortor eu turpis pharetra consequat quis non metus. Aliquam aliquam, orci eu suscipit pellentesque, mauris dui tincidunt enim, eget iaculis ante dolor non turpis.
          </p>
          <p>
            Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur. At vero eos et accusamus et iusto odio dignissimos ducimus.
          </p>
        </div>
        <a
          className="btn btn-cta-secondary"
          href="http://themes.3rdwavemedia.com/website-templates/responsive-bootstrap-theme-for-products-velocity/"
          target="_blank"
        >
          <i
            className="fa fa-thumbs-o-up"
          />
           Back my project
        </a>
      </div>
      <hr
        className="divider"
      />
    </div>
  </div>
</section>
`;
